server/services/prismic.js

```diff
@@ -167,3 +165,136 @@ export async function getCuratedList(id: string) {
 
   return curatedList;
 }
+
+async function getExhibitions(page:number = 1, pageSize:number = 40): Promise<any> {
+  const prismic = await getPrismicApi();
+  const exhibitionsList = await prismic.query([
+    Prismic.Predicates.any('document.type', ['exhibitions'])
+  ], { orderings: '[my.exhibitions.start desc]', page, pageSize });
+  return exhibitionsList;
+}
+
+async function createPromos(allResults, type) {
+  switch (type) {
+    case 'exhibition':
+      return await createExhibitionPromos(allResults);
+    case 'event':
+      return await createEventPromos(allResults);
+  }
+}
+
+async function createExhibitionPromos(allResults: Object): Promise<Array<ExhibitionPromo>> {
+  const allPromos = allResults.results.map((e):ExhibitionPromo => {
+    return {
+      id: e.id,
+      url: `/exhibitions/${e.id}`,
+      title: asText(e.data.title),
+      image: prismicImage(e.data.promo[0].primary.image),
+      description: asText(e.data.promo[0].primary.caption),
+      start: e.data.start ? e.data.start : '2007-06-21T00:00:00+0000',
+      end: e.data.end
+    };
+  });
+
+  const permanentPromos = allPromos.filter((e) => {
+    return !e.end;
+  });
+
+  const temporaryPromos = allPromos.filter((e) => {
+    return e.end;
+  });
+
+  return permanentPromos.concat(temporaryPromos);
+}
+
+async function createEventPromos(allResults): Promise<Array<EventPromo>> {
+  return allResults.results.map((event): EventPromo => {
+    const promo = event.data.promo[0];
+    const promoImage = promo && promo.primary.image;
+    const promoCaption = promo && promo.primary.caption;
+
+    return event.data.times.map(eventAtTime => {
+      return {
+        id: event.id,
+        title: asText(event.data.title),
+        url: `/events/${event.id}`,
+        start: eventAtTime.startDateTime,
+        end: eventAtTime.endDateTime,
+        image: prismicImage(promoImage),
+        description: asText(promoCaption)
+      };
+    });
+  }).reduce((acc, curr) => {
+    return curr.concat(acc);
+  }, []).sort((a, b) => {
+    return getNumberFromString(b.start || '') - getNumberFromString(a.start || '');
+  });
+}
+
+function getNumberFromString(string: string): number {
+  return Number(string.replace(/\D/g, ''));
```

Comment:
What type of input are we expecting here?
These are dates aren't they?

Reply:
strings (of dates, in the form '12-01-2017 07:45:00 pm +00:00')

Reply:
SO then that would give us:
`120120170745000000`

It would ignore PM et, which is weird.

As these are JS dates, you can compare on them anyway.
e.g.
```JS
new Date('12-01-2017 07:45:00 pm +00:00') > new Date('12-01-2017 07:46:00 pm +00:00') // <= false
```